native/xqlitenif/src/r2d2.rs

```diff
@@ -0,0 +1,252 @@
+use crate::atoms::{
+    cannot_allocate_binary, cannot_encode, cannot_execute, cannot_fetch_row,
+    cannot_open_database, cannot_prepare_statement, connection_not_found, timeout,
+    unsupported_value,
+};
+use dashmap::DashMap;
+use r2d2::Pool;
+use r2d2_sqlite::SqliteConnectionManager;
+use rustler::types::atom::{error, nil, ok};
+use rustler::{Encoder, Env, OwnedBinary, Resource, ResourceArc, Term};
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::OnceLock;
+
+type XqlitePool = Pool<SqliteConnectionManager>;
+type XqlitePools = DashMap<u64, XqlitePool>;
+
+#[derive(Debug)]
+pub(crate) struct XqliteConn(u64);
+impl Resource for XqliteConn {}
```

Comment:
```suggestion
#[resource_impl]
impl Resource for XqliteConn {}
```

Reply:
btw, should have also included `use rustler::resource_impl;` at the top. I have included it here: https://github.com/dimitarvp/xqlite/pull/9/commits/fbe28dc57cac206ce1ce469128f68629534c8e4b